processing/src/main/java/io/druid/query/ChainedExecutionQueryRunner.java

```diff
@@ -70,87 +74,113 @@ public ChainedExecutionQueryRunner(
+                      public Iterable<T> call()
+                      {
+                        try {
+                          Sequence<T> result = queryRunner.run(threadSafeQueryPlus, responseContext);
+                          if (result == null) {
+                            throw new ISE("Got a null result! Segments are missing!");
                           }
 
-                          return exec.submit(
-                              new AbstractPrioritizedCallable<Iterable<T>>(priority)
-                              {
-                                @Override
-                                public Iterable<T> call()
-                                {
-                                  try {
-                                    Sequence<T> result = input.run(threadSafeQueryPlus, responseContext);
-                                    if (result == null) {
-                                      throw new ISE("Got a null result! Segments are missing!");
-                                    }
-
-                                    List<T> retVal = result.toList();
-                                    if (retVal == null) {
-                                      throw new ISE("Got a null list of results! WTF?!");
-                                    }
-
-                                    return retVal;
-                                  }
-                                  catch (QueryInterruptedException e) {
-                                    throw Throwables.propagate(e);
-                                  }
-                                  catch (Exception e) {
-                                    log.error(e, "Exception with one of the sequences!");
-                                    throw Throwables.propagate(e);
-                                  }
-                                }
-                              }
-                          );
+                          List<T> retVal = result.toList();
+                          if (retVal == null) {
+                            throw new ISE("Got a null list of results! WTF?!");
+                          }
+
+                          return retVal;
+                        }
+                        catch (QueryInterruptedException e) {
+                          throw Throwables.propagate(e);
+                        }
+                        catch (Exception e) {
+                          log.error(e, "Exception with one of the sequences!");
+                          throw Throwables.propagate(e);
                         }
-                    )
-                )
+                      }
+                    }
+                ).map(exec::submit)
             );
 
             queryWatcher.registerQuery(query, futures);
 
             try {
+              final DateTime deadline =
+                  QueryContexts.hasTimeout(query) ?
+                  DateTimes.utc(DateTimes.nowUtc().getMillis() + QueryContexts.getTimeout(query)) :
```

Comment:
`DateTimes.nowUtc().plusMillis(QueryContexts.getTimeout(query))` is better

Reply:
fixed